benchmarks/bench_kernel_pca_solvers_comparison1.py

```diff
@@ -0,0 +1,153 @@
+"""
+=========================================
+Kernel PCA Solvers comparison benchmark 1
+=========================================
+
+This benchmark shows that the approximate solvers provided in Kernel PCA can
+help drastically improve its execution speed when an approximate solution
+(small `n_components`) is acceptable. In many real-world datasets a few
+hundreds of principal components are indeed sufficient enough to capture the
+underlying distribution.
+
+Description:
+------------
+An increasing number of examples is used to train a KernelPCA, between
+`min_n_samples` (default: 101) and `max_n_samples` (default: 4000) with
+`n_samples_grid_size` positions (default: 4). For each training sample size,
+KernelPCA models are trained for the various possible `eigen_solver` values.
+All of them are trained to obtain p principal components (default: 100).
+The execution times are displayed in a plot at the end of the experiment.
+
+What you can observe:
+---------------------
+When the number of samples provided gets large, the dense solver takes a lot
+of time to complete, while the randomized method returns similar results in
+much shorter execution times.
+
+Going further:
+--------------
+You can increase `max_n_samples` and `nb_n_samples_to_try` if you wish to
+explore a wider range of values for `n_samples`.
+
+You can also set `include_arpack=True` to add this other solver in the
+experiments (much slower).
+
+Finally you can have a look at the second example of this series,
+"Kernel PCA Solvers comparison benchmark 2", where this time the number of
+examples is fixed, and the desired number of components varies.
+"""
+# Author: Sylvain MARIE
+
+from datetime import datetime
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.decomposition import KernelPCA
+from sklearn.datasets import make_circles
+
+
+print(__doc__)
+
+
+# 1- Design the Experiment
+# ------------------------
+min_n_samples, max_n_samples = 101, 4000  # min and max n_samples to try
+n_samples_grid_size = 4                   # nb of positions in the grid to try
+# generate the grid
+n_samples_range = [min_n_samples + np.floor((x / (n_samples_grid_size - 1))
+                                            * (max_n_samples - min_n_samples))
+                   for x in range(0, n_samples_grid_size)]
+
+n_components = 100      # the number of principal components we want to use
+n_iter = 3              # the number of times each experiment will be repeated
+include_arpack = False  # set this to True to include arpack solver (slower)
+
+
+# 2- Generate random data
+# -----------------------
+np.random.seed(0)
+n_features = 2
```

Comment:
I'm quite confused, does it really make sense to user kernelPCA to go from 2 features to 100? Is this a thing people do?

Reply:
Not sure I understand this comment. The number of features is fixed, this line cannot be modified. It is just for readability. Should I remove it ?